mozilla_django_oidc/views.py

```diff
@@ -114,15 +114,15 @@ def redirect_url(self):
 
     def dispatch(self, request, *args, **kwargs):
         """Log out the user."""
-
-        if request.user.is_authenticated():
+        # Check if a method exists to build the URL to log out the user
+        # from the OP.
+        logout_from_op = import_from_settings('OIDC_OP_LOGOUT_URL_METHOD', '')
```

Comment:
I believe the simplest solution would be option 4. Logout the user from the OP only if the setting is present and the user is logged in and redirect to the `self.redirect_url` in all other cases.

Another option could be:
```python
if not user.is_authenticated() and import_from_settings('OIDC_OP_LOGOUT_URL_METHOD', ''):
    return HttpResponseBadRequest('already signed out')
```
This way we won't bypass the setting and at the same time we can silently redirect to `self.redirect_url` when the setting is missing.